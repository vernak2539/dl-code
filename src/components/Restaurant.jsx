import React from 'react';

export default ({ name, tags, price, image }) => {
	const priceInPoundSymbols = '£'.repeat(price); // doesn't work in IE for the moment
	const categoriesAndPrice = [...tags, priceInPoundSymbols];

	const restaurantImageStyles = {
		backgroundImage: `url(${image})`
	};

	return (
		<div className="restaurant">
			<div className="restaurant-image" style={restaurantImageStyles} />
			<h2>{name}</h2>
			<div className="categories-price text-gray">{categoriesAndPrice.join(' • ')}</div>
		</div>
	);
};
