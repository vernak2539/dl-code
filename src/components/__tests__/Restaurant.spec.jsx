import React from 'react';
import { render } from 'enzyme';
import Restaurant from '../Restaurant';

describe('Component > Restaurant', () => {
	it('should render correctly', () => {
		const restaurant = {
			id: '63460',
			name: 'Burrito Kitchen',
			image:
				'https://f.roocdn.com/images/menus/25785/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1477308353',
			url: 'https://deliveroo.co.uk/menu/london/bank/burrito-kitchen-cheapside',
			price: 2,
			tags: ['Mexican', 'Burritos', 'Halal']
		};

		expect(render(<Restaurant {...restaurant} />)).toMatchSnapshot();
	});
});
