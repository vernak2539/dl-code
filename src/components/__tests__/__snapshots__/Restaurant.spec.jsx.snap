// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Restaurant should render correctly 1`] = `
<div
  class="restaurant"
>
  <div
    class="restaurant-image"
    style="background-image:url(https://f.roocdn.com/images/menus/25785/header-image.jpg?width=360&height=200&auto=webp&format=jpg&fit=crop&v=1477308353)"
  >
    
  </div>
  <h2>
    Burrito Kitchen
  </h2>
  <div
    class="categories-price text-gray"
  >
    Mexican • Burritos • Halal • ££
  </div>
</div>
`;
